backend/src/database/state-migrations/impossible-migration.ts

```diff
@@ -0,0 +1,14 @@
+import { RestoreError } from '../../utils/restore-error';
+import type { Migrations } from './migrations';
+
+export const impossibleMigration: Migrations = {
+    database: (_entityManager, exerciseId) => {
+        throw new RestoreError('The migration is not possible', exerciseId);
+    },
+    stateExport: (stateExport) => {
+        throw new RestoreError(
+            'The migration is not possible',
+            stateExport.currentState.id ?? 'unknown id'
+        );
+    },
```

Comment:
Is it really necessary to have two different migration functions?

Would it be possible to do this with only one, by exporting the stuff saved in the database and migrating it with the same function to the new state?

The problem I currently see is that nobody is going to write migrations because they are too much work. -> Premature optimization.

Reply:
I've completely reworked it now. It now uses two methods, one for states and one for actions. With this, I'm handling the loading/writing to the database in the util method, and people who write migrations only have to write the absolute minimum. Do you like this approach better?